src/kakarot/instructions/system_operations.cairo

```diff
@@ -234,33 +222,27 @@ namespace SystemOperations {
         range_check_ptr,
         bitwise_ptr: BitwiseBuiltin*,
     }(ctx: model.ExecutionContext*) -> model.ExecutionContext* {
+        // See https://docs.cairo-lang.org/0.12.0/how_cairo_works/functions.html#retrieving-registers
+        alloc_locals;
+        let fp_and_pc = get_fp_and_pc();
+        local __fp__: felt* = fp_and_pc.fp_val;
         let sub_ctx = CallHelper.init_sub_context(
-            calling_ctx=ctx, with_value=TRUE, read_only=ctx.read_only
+            ctx=ctx, with_value=TRUE, read_only=ctx.call_context.read_only, self_call=FALSE
         );
-        if (ctx.read_only * sub_ctx.call_context.value != FALSE) {
+        if (ctx.call_context.read_only * sub_ctx.call_context.value != FALSE) {
             let (revert_reason_len, revert_reason) = Errors.stateModificationError();
-            let ctx = ExecutionContext.revert(ctx, revert_reason, revert_reason_len);
+            let ctx = ExecutionContext.stop(ctx, revert_reason_len, revert_reason, TRUE);
             return ctx;
         }
 
-        if (sub_ctx.call_context.value == 0) {
-            return sub_ctx;
-        }
-
-        let (native_token_address_) = native_token_address.read();
-        let amount_u256 = Helpers.to_uint256(sub_ctx.call_context.value);
-        let sender = ctx.starknet_contract_address;
-        let recipient = sub_ctx.starknet_contract_address;
-        let (success) = IERC20.transferFrom(
-            contract_address=native_token_address_,
-            sender=sender,
-            recipient=recipient,
-            amount=amount_u256,
+        let (value_high, value_low) = split_felt(sub_ctx.call_context.value);
```

Comment:
why isn't CallContext.value a uint256? 
Though I agree it's unlikely someone will be able to input this amount

Reply:
good question, probably legacy we need to fix

Reply:
I don't fix this in this PR because it's not related and in any case very unlickely that a value be greater than 2**251 ETH, but still we should fix this I agree

Reply:
#773